magentov2/plugin/Indodana/PayLater/Helper/Transaction.php

```diff
@@ -0,0 +1,514 @@
+<?php
+
+namespace Indodana\PayLater\Helper;
+use IndodanaCommon\IndodanaInterface;
+use IndodanaCommon\IndodanaCommon;
+use IndodanaCommon\IndodanaConstant;
+use Magento\Framework\App\Helper\AbstractHelper;
+use \Magento\Framework\App\Filesystem\DirectoryList;
+
+class Transaction extends AbstractHelper implements IndodanaInterface
+{
+  private $indodanaCommon;
+  protected $_helper;
+  protected $_customer;
+  protected $_urlInterface;
+  protected $_dir;
+  public function __construct(
+    Data $helper,
+    \Magento\Customer\Api\CustomerRepositoryInterface $customerRepositoryInterface,
+    \Magento\Framework\UrlInterface $urlInterface,
+    \Magento\Framework\App\Filesystem\DirectoryList $directoryList
+    )
+  {
+          $this->_helper = $helper;
+          $this->_customer= $customerRepositoryInterface;
+          $this->_urlInterface = $urlInterface;
+          $this->_dir = $directoryList;
+          require_once($this->_dir->getPath(DirectoryList::APP). '/code/Indodana/PayLater/autoload.php' );
+  }
+
+  public function getIndodanaCommon()
+  {
+    if (!isset($this->indodanaCommon)) {
+      $apiKey = $this->_helper->getApiKey();
+      $apiSecret = $this->_helper->getApiSecret();
+      $environment = $this->_helper->getEnvironment();
+
+      $this->indodanaCommon = new IndodanaCommon([
+        'apiKey'        => $apiKey,
+        'apiSecret'     => $apiSecret,
+        'environment'   => $environment,
+        'seller'        => $this->getSeller()
+      ]);
+    }
+    return $this->indodanaCommon;
+  }
+
+  public function getTotalAmount($order)
+  {
+    return (float) $order->getGrandTotal();
+  }
+
+  public function getTotalDiscountAmount($order)
+  {
+    $totalDiscountAmount = 0;
+    foreach ($order->getAllVisibleItems() as $item) {
+      $totalDiscountAmount += (float) $item->getDiscountAmount();
+    }
+    return $totalDiscountAmount;
+  }
+
+  public function getTotalShippingAmount($order)
+  {
+    // I'm not really sure whether to use getShippingInclTax() or getShippingAmount()
+    // For get installment options
+    $totalShippingAmount = (float) $order->getShippingAddress()->getShippingInclTax();
+    // For checkout
+    if (!$totalShippingAmount) {
+      $totalShippingAmount = (float) $order->getShippingInclTax();
+    }
+    return $totalShippingAmount;
+  }
+
+  public function getTotalTaxAmount($order)
+  {
+    $totalTaxAmount = 0;
+    foreach ($order->getAllVisibleItems() as $item) {
+      $totalTaxAmount += (float) $item->getTaxAmount();
+    }
+    return $totalTaxAmount;
```

Comment:
Please add a newline before `return` statement

Instead of
```
foreach ($order->getAllVisibleItems() as $item) {
    $totalTaxAmount += (float) $item->getTaxAmount();
}
return $totalTaxAmount;
```

This is the correct one
```
foreach ($order->getAllVisibleItems() as $item) {
    $totalTaxAmount += (float) $item->getTaxAmount();
}

return $totalTaxAmount;
```

Please apply this on all changes on this PR

Reply:
DONE

Reply:
done